src/index.js

```diff
@@ -1,5 +1,6 @@
 /* global process */
 /* eslint no-console: 0 */
+/* eslint no-process-exit: 0 */
```

Comment:
@luftywiranda13 I'm not sure about this one. Any opinions?

Reply:
The recommended approach is to throw the error itself. This can be done in all three instances.

From https://eslint.org/docs/rules/no-process-exit:
> By throwing an error in this way, other parts of the application have an opportunity to handle the error rather than stopping the application altogether. If the error bubbles all the way up to the process without being handled, then the process will exit and a non-zero exit code will returned, so the end result is the same.

Reply:
I'm not sure about if throwing will also print the stack trace which I'd want to avoid. I'd propose to merge this one and then tackle the warning/recommendation in a separate PR.

Reply:
> There may be a part of a Node.js application that is responsible for determining the correct exit code to return upon exiting. In that case, you should turn this rule off to allow proper handling of the exit code.

I think that's exactly the case here.

Reply:
yeah, throw error if we want to handle or do something about the error. if not, I think it's ok to just exit non-zero

Reply:
As far as I can see, its `process.exit(1)` in three places after catching an error. While you might want to tweak the error message, throwing an error will have the same effect:

> If the error bubbles all the way up to the process without being handled, then the process will exit and a non-zero exit code will returned, so the end result is the same.

While I don't see disabling the rule as too much of a problem, I generally respect eslint(like the wise big brother with real life experience) and prefer following that. Though this might not be a real use case now, if anybody was invoking `lint-staged` programmatically, `throw err` vs `process.exit` becomes relevant.

BTW,  `eslint-plugin-unicorn` has a similar rule with tweaks for bin targets - https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/no-process-exit.md. And this allows for use of `process.exit`.